lib/ably/realtime/channel.rb

```diff
@@ -179,15 +184,26 @@ def presence
         @presence
       end
 
-      # Return the message history of the channel.
+      # Return the message history of the channel
+      #
+      # Once attached to a channel, you can retrieve messages published on the channel before the
```

Comment:
It is different in that what you describe as an argument error is going to fail only on certain occasion.

```ruby
# this fails all the time because you have not specified `id`
get_thing

# this may fail or not depending on the channel state
channel.history(until_attach: true)

# this never fails because it is just ignored
channel.history(something: 'hello')
```